### ISSUE TITLE
string enum member used as "computed enum member" types as number.

### ISSUE THREAD

**Issue by tadhgmister** [microsoft/TypeScript#40862]
**TypeScript Version:**  all the ones in playground, latest is v4.1.0-dev20201001

**Search Terms:**

enum, incorrect type, computed enum, number, string

**Code**

```ts
enum Constants {
    hello = "Hello"
}
enum Potato {
    // THIS IS CONSIDERED A NUMERIC ENUM!!!!!
    h = Constants.hello
}
// Error: Type 'Potato' is not assignable to type 'string'.(2322)
const value: string = Potato.h;

// intellisense shows Potato.h = "Hello", but tsc compiles as though it's a number
console.log(Potato.h.toExponential());
```

**Expected behavior:**

Using a string enum member as the computed key in another enum should either be considered a string enum or be not allowed.

**Actual behavior:**

`Potato.h` is typed as number despite being assigned to a string. `Potato.h.toExponential()` passes type checking and throws runtime error.

[**Playground Link**](https://www.typescriptlang.org/play?ts=4.1.0-dev.20200930#code/KYOwrgtgBAwg9iAzgFwIYmYqBvAsAKCiKgAtgAbcuKAXigCIAJCq+ggXwNEigAU40yankLEA9GKgAVRgEkAylAWwA8gDl5sgCIBRAEo6tUAIJQ1AVQCy+2TCg6LlgIQuXBYqVqwEKdJgB0ZJRwHAQS0gCeAA7AUADk-IJwcVAAllggAlCoiIipAOYgqABG5LFCUMjRsXEoAE6pIPlx-gAUAEwAzO3tAJQEAMY+yFAAbqjkYMAAXFD1jfleiahCgQDcBGGSjcgs6aCIsYgkcADuWMurJF5MLHD0ADRQxWAjmANQQxBRqWVYOZUTmB8tdUshatkoOAIMVgHVBj44GV-FR8q1LnBAv4hDoAB5RBCgZCpCatXq9Db4IA)

**Related Issues:** 

https://stackoverflow.com/questions/64145849/how-can-i-pass-an-enum-with-keys-of-another-string-enum-into-a-function-that-a

**Comment by jcalz** [comment]
This behavior is 🤪 

Found my way here after researching [a different SO question](https://stackoverflow.com/q/70885470/2887218).

**Comment by ahejlsberg** [comment]
Fixed by #50528.
